### DISCUSSION TITLE
Is it still necessary to `import isNill from lodash.isNill` instad of `import {isNil} from lodash`?

### DISCUSSION THREAD

**Discussion by seyyedhassanpour** [lodash/lodash#5742] [Q&A] [answered]
why we need to use
```js
import groupBy from 'lodash.groupby'  
import isNil from 'lodash.isnil'
```
instead of below?
```js
import {groupBy,isNil} from 'lodash'
```
is this still needed?

There are a bunch of [different approaches and results](https://stackoverflow.com/questions/35250500/correct-way-to-import-lodash) and most of the are for 2020, 2019 or older.
What is the rightway? can we use a syntax like this 
```
import {groupBy,isNil} from 'lodash' or 'lodash-es'
```

And what is the difference between the lodash` and `lodash-es`?
It would  greatly appreciated it if anyone shed some light on this topic.

**Comment by jdalton** [comment, marked as answer]
For modern environments you should import from lodash-es.
```js
import { groupBy } from 'lodash-es'
```

**Reply by seyyedhassanpour** [reply]
Thank you for your response. It would be greatly appreciated if you answer this one:
With this pattern, all the package wont be included into final bundle?

**Reply by ZachHaber** [reply]
Assuming that none of your other dependencies rely on the base 'lodash' package, that's correct. If they do, however, you will likely get double lodash included.

**Reply by hellohejinyu** [reply]
> Assuming that none of your other dependencies rely on the base 'lodash' package, that's correct. If they do, however, you will likely get double lodash included.

```js
import groupBy from 'lodash/groupBy'
```

What will happen if I use it like this?🧐

**Reply by ZachHaber** [reply]
> What will happen if I use it like this?🧐

Then you'll get double `groupBy` if you have the full `lodash` imported anywhere already... Triple if you also have `lodash-es`, and quadruple if you also have `lodash.groupBy`... etc... 😢

There are so many different way to accidentally end up with multiple lodash packages installed.
